security/oc_cred.c

```diff
@@ -552,13 +557,22 @@ oc_sec_add_new_cred(size_t device, bool roles_resource, oc_tls_peer_t *client,
       uint8_t key[64];
       memcpy(key, privatedata, privatedata_size);
       int key_size = oc_base64_decode(key, privatedata_size);
-      if (key_size < 0) {
+      if (key_size != SYMMETRIC_KEY_128BIT_LEN &&
```

Comment:
Maybe a helper function for the key size check and use it here and on lines 570-571.

Reply:
fixed